### PR TITLE
:sparkles: Changed jsonScorecardResultV2 type Public

### PR DESCRIPTION
**Pull request by naveensrinivasan**
#### What kind of change does this PR introduce?

Bug fix

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
- Fixes https://github.com/ossf/scorecard/issues/1673

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
Made the jsonScorecardResultV2 to public to make it easier for the consumers of Scorecard.
```

### CODE REVIEW

File: pkg/json.go [hunk 1]
```diff
@@ -72,7 +72,8 @@ func (s jsonFloatScore) MarshalJSON() ([]byte, error) {
 }
 
 //nolint:govet
```
Comment (justaugustus) [inline pkg/json.go:74]:
```suggestion
```

the `nolint` is probably no longer required

**Review by justaugustus — CHANGES_REQUESTED** [review]
@naveensrinivasan -- One nit about the `nolint`!

**Review by azeemshaikh38 — APPROVED** [review]

**Review by laurentsimon — APPROVED** [review]
Thanks!

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan -- One nit about the `nolint`!

@justaugustus  Looks like that is required for fieldalingment. Can you please 👀 ?

**Review by justaugustus — APPROVED** [review]
> @justaugustus Looks like that is required for fieldalingment. Can you please 👀 ?

@naveensrinivasan -- Yep, I think moving the `date` field(str) to the bottom would fix it, but that's non-blocking!

**Comment by naveensrinivasan** [comment]
> > @justaugustus Looks like that is required for fieldalingment. Can you please 👀 ?
> 
> @naveensrinivasan -- Yep, I think moving the `date` field(str) to the bottom would fix it, but that's non-blocking!

👍

**Pull request merged**